drivers/display/ssd1306.c

```diff
@@ -408,7 +408,13 @@ static int ssd1306_init(const struct device *dev)
 	}
 
 	if (config->reset.port) {
-		gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_INACTIVE);
+		int ret;
+
+		ret = gpio_pin_configure_dt(&config->reset,
+					    GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
```

Comment:
Treat anything other than 0 (successful) as error.

Reply:
As I already mentioned [here](https://github.com/zephyrproject-rtos/zephyr/pull/38021#discussion_r696761297) this is not something that should be done per function documentation:

```
 * @retval 0 If successful.
 * @retval -ENOTSUP if any of the configuration options is not supported
 *                  (unless otherwise directed by flag documentation).
 * @retval -EINVAL Invalid argument.
 * @retval -EIO I/O error when accessing an external GPIO chip.
 * @retval -EWOULDBLOCK if operation would block.
```

So `ret < 0` is perfectly correct, and I'm not willing to change it.